Homework 1/ArrayWithUniqueElements.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Homework_1
+{
+    internal class ArrayWithUniqueElements
+    {
+        private int[] array;
+
+        public ArrayWithUniqueElements(int[] inputArray)
+        {
+            array = FillArrayWithUniqueElements(inputArray);
+        }
+
+        public int[] Array { get => array; }
+
+        private int[] FillArrayWithUniqueElements(int[] inputArray)
+        {
+            var array = new int[1] { inputArray[0] };
+            for (int i = 1; i < inputArray.Length; i++)
+            {
+                if (!ContainsElement(inputArray[i], array))
+                {
+                    array = AppendElementToArray(inputArray[i], array);
+                }
+            }
+            return array;
+        }
+
+        private int[] AppendElementToArray(int newElement, int[] array)
+        {
```

Comment:
Not efficient, you re-create array each time, much more efficient to create array once and after whole processing resize it